### ISSUE TITLE
Unable to use path with trailing slash

### ISSUE THREAD

**Issue by sempasha** [byu-oit/openapi-enforcer#22]
When path in openapi definition has a trailing slash - it's unable to call method associated with this path.
The reason is ambiguous behaviour:

  1. Trailing slashes won't removed while openapi definition parsing;
  2. Trailing slashes will removed while request parsing, when enforcer search matching controller method ([1](https://github.com/byu-oit/openapi-enforcer/blob/cebf9643cca837c49b37c26e2921a8f2e1e2af1f/src/enforcers/OpenApi.js#L40), [2](https://github.com/byu-oit/openapi-enforcer/blob/cebf9643cca837c49b37c26e2921a8f2e1e2af1f/src/enforcers/OpenApi.js#L93), [3](https://github.com/byu-oit/openapi-enforcer/blob/9f2974118065926b5f44705470c7ff4c5d10d4c2/src/enforcers/Paths.js#L147)).

Sample service definition (openapi.yaml):
```yaml
openapi: 3.0.2

servers:
  - url: http://example.com

info:
  description: Test API
  version: 0.0.0
  title: Test API Definition
  contact:
    name: example.com
    email: user@example.com
    url: http://example.com
  license:
    name: UNLICENSED

paths:

  "/some":
    post:
      operationId: postSome
      responses:
        200:
          content:
            application/json:
              schema:
                type: string
          description: Save some data
      summary: Save some object
    x-controller: test

  "/other/":
    post:
      operationId: postOther
      responses:
        200:
          content:
            application/json:
              schema:
                type: string
          description: Save other data
      summary: Save other object
    x-controller: test

components: {}
```
Sample service (index.js)
```js
const Enforcer = require('openapi-enforcer-middleware');
const express = require('express');
const http = require('http');
const path = require('path');

let app, enforcer, server;

enforcer = Enforcer(path.resolve(__dirname, 'openapi.yaml'));
enforcer.controllers({
    test: {
        postSome: function (req, res) {
            res.status(200).send('postSome');
        },
        postOther: function (req, res) {
            res.status(200).send('postOther');
        }
    }
});
app = express();
server = http.createServer(app);
app.use(enforcer.middleware());
server.listen(3000, error => {
    if (error) {
        console.error(error);
    } else {
        console.log('listening on port 3000');
    }
});
```

Start service:
```bash
$ node ./index.js
```

Perform requests:
```bash
$ curl -X POST http://localhost:3000/some  
postSome

$ curl -X POST http://localhost:3000/other
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>Cannot POST /other</pre>
</body>
</html>

$ curl -X POST http://localhost:3000/other/
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>Cannot POST /other/</pre>
</body>
</html>
```

Expected behaviour is:
```bash
$ curl -X POST http://localhost:3000/some  
postSome

$ curl -X POST http://localhost:3000/other
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>Cannot POST /other</pre>
</body>
</html>

$ curl -X POST http://localhost:3000/other/
postOther
```

Seems there is no need to do [`utils.edgeSlashes`](https://github.com/byu-oit/openapi-enforcer/blob/fe04760789f1bf546c81ace4975accd4fa606657/src/util.js#L119) while path normalization at all.

**Comment by Gi60s** [comment]
That is such a curious behavior it makes me LOL.

The solution does seem to be removing the path normalization, as you stated, but doing so may lead to errors for those who already have their API's being used.

Here is what I'm going to do:

1. Create an option that can be used to skip path normalization.
2. Produce warning messages if some paths end with slashes and some do not.
3. Produce warning messages if path normalization is enabled and some paths end in slashes.
4. Produce error messages if path normalization is disabled and two nearly identical paths exist. For example `/foo` and `/foo/` is fine if you have path normalization disabled, but it's an error if you have it enabled because you can only reach the first path.

**Comment by sempasha** [comment]
What kind of issues you are expect for those who already use openapi-enforcer?

  1. For those who has no trailing slash nothing will change.
  2. For those who has trailing slashes ... who use trailing slashes when it doesn't work?

So I think changes won't break things, only fix it.
What about option, I think path normalization is internal feature of openapi-enforcer, can't image how developers can use it.

**Comment by Gi60s** [comment]
I'm referring to user's of the API who may have used bad practices to access the API. For example, if your path defines `/foo` and they have been requesting `/foo/` then the user of your API will no longer be able to access it. Although your API is not technically broken by the change, services that use your API could be affected.

**Comment by Gi60s** [comment]
Fixed in PR #23 and published to NPM on version `1.4.2`.

Thanks for reporting this and taking the time to understand the problem. That made it an easy fix.
